tests/test_dataset.py

```diff
@@ -711,3 +711,44 @@ def test_select_mtzdtype_ValueError(data_merged, dtype):
     """
     with pytest.raises(ValueError):
         data_merged.select_mtzdtype(dtype)
+
+
+@pytest.mark.parametrize("merged", [True, False])
+@pytest.mark.parametrize("hkl_as_ds", [True, False])
+@pytest.mark.parametrize("range_index", [True, False])
+def test_hkls_property_setter(
+    data_merged, data_unmerged, merged, hkl_as_ds, range_index
+):
+    """
+    Test the setter for the .hkls property of rs datasets
+    """
+    if merged:
+        input_ds = data_merged
+    else:
+        input_ds = data_unmerged
+
+    ds = input_ds.copy()
+
+    if range_index:
+        ds = ds.reset_index()
+
+    hmax = 20
```

Comment:
We have two pytest fixtures (`hkls` and `dataset_hkl`) that could probably support this test, rather than rolling your own. If the other two are insufficient, maybe we should modify them so that this can be used elsewhere as well?

Reply:
i don't think these fixtures are appropriate, because the hkls in this case should be matched to the length of the dataset. it might make sense to use them to test the case where the dataset length and hkl length differ.

Reply:
maybe we could match these hkls to those in data_merged and/or data_unmerged?

Reply:
okay -- i changed the hkl fixtures to be matched to the data fixtures. let me know if you think this is a good solution.